providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java

```diff
@@ -123,8 +127,27 @@ static void observeEventStream(final BlockingQueue<StreamPayload> writeTo,
         log.info("Initializing sync stream observer");
 
         while (!shutdown.get()) {
+            Exception metadataException = null;
             log.debug("Initializing sync stream request");
-            serviceStub.syncFlags(request, new GrpcStreamHandler(streamReceiver));
+            final SyncFlagsRequest.Builder syncRequest = SyncFlagsRequest.newBuilder();
+            final GetMetadataRequest.Builder metadataRequest = GetMetadataRequest.newBuilder();
+            Map<String, Object> metadata = Collections.emptyMap();
+
+            if (selector != null) {
+                syncRequest.setSelector(selector);
+            }
+
+            serviceStub.syncFlags(syncRequest.build(), new GrpcStreamHandler(streamReceiver));
+            try {
+                GetMetadataResponse metadataResponse = serviceBlockingStub.getMetadata(metadataRequest.build());
+                metadata = convertProtobufMapToStructure(metadataResponse.getMetadata().getFieldsMap()).asObjectMap();
+            } catch (Exception e) {
+                // the chances this call fails but the syncRequest does not are slim
+                // it could be that the server doesn't implement this RPC
+                // instead of logging here, retain the exception and only log if the
+                // streamReceiver doesn't error
+                metadataException = e;
+            }
```

Comment:
This is the meat of the new change... we call the getMetadata RPC as soon as we start the stream.

_IF_ the sync-metadata somehow fails but the stream still connects, we log it below; but this shouldn't happen since both use the same underlying channel.